src/sentry/lang/native/minidump.py

```diff
@@ -180,3 +194,73 @@ def frames_from_minidump_thread(thread):
         'package': frame.module.code_file if frame.module else None,
         'trust': frame.trust,
     } for frame in reversed(list(thread.frames()))]
+
+
+def get_attached_minidump(data):
+    cache_key = cache_key_for_event(data)
+    attachments = attachment_cache.get(cache_key) or []
+    return next((a for a in attachments if a.type == MINIDUMP_ATTACHMENT_TYPE), None)
+
+
+def merge_symbolicator_minidump_response(data, response):
+    sdk_info = get_sdk_from_event(data)
+
+    # TODO(markus): Add OS context here when `merge_process_state_event` is no
+    # longer called for symbolicator projects
+
+    data.setdefault('debug_meta', {})['images'] = images = []
+
+    for complete_image in response['modules']:
+        image = {}
+        merge_symbolicator_image(
+            image, complete_image, sdk_info,
+            lambda e: handle_symbolication_failed(e, data=data)
+        )
+        images.append(image)
+
+    setdefault_path(data, 'threads', 'values', value=[])
+    data_threads = get_path(data, 'threads', 'values')
+    assert isinstance(data_threads, list)
+
+    data_threads_by_id = {
```

Comment:
Can't we just replace threads rather than matching them based on their id? This could have already failed in the old code but there was no better way to do this. Now, you could just overwrite what's there. I would base it slightly more off what we do in `wip/symbolicator`. 

This also applies to the frame merging code below, as well as generally to `reprocess_minidump`.

Reply:
Yeah we just replace threads now. There is no way to overwrite stacktraces anymore at all.